File: ObjectAvoidance/java/src/main/java/elite/math/trig/Cosine.java
package elite.math.trig;

import elite.math.trig.triangle.Adjacent;
import elite.math.trig.triangle.Hypotenuse;

/**
 *
 * @author CFerg (Elite)
 */
public class Cosine {
    
    /**
     *
     */
    public double cosine;
    
    /**
     *
     * @param a
     * @param h
     */
    public Cosine(Adjacent a, Hypotenuse h){
        this.cosine = (a.adjacent / h.hypotenuse);
    }
    
    /**
     *
     * @param arccos
     */
    public Cosine(Arccosine arccos){
        this.cosine = (1 / arccos.arccosine);
    }
    
    /**
     *
     * @param sec
     */
    public Cosine(Secant sec){
        this.cosine = (1 / sec.secant);
    }
}

File: ObjectAvoidance/kotlin/src/elite/math/trig/Cosine.kt
package elite.math.trig

import elite.math.trig.triangle.Adjacent
import elite.math.trig.triangle.Hypotenuse

/**
 *
 * @author CFerg (Elite)
 */
class Cosine {

    /**
     *
     */
    var cosine: Double = 0.toDouble()

    /**
     *
     * @param a
     * @param h
     */
    constructor(a: Adjacent, h: Hypotenuse) {
        this.cosine = a.adjacent / h.hypotenuse
    }

    /**
     *
     * @param arccos
     */
    constructor(arccos: Arccosine) {
        this.cosine = 1 / arccos.arccosine
    }

    /**
     *
     * @param sec
     */
    constructor(sec: Secant) {
        this.cosine = 1 / sec.secant
    }
}